src/linear_algebra/basic.lean

```diff
@@ -1029,6 +1029,20 @@ begin
   rw [range_comp, submodule.map_neg, submodule.map_id],
 end
 
+/-- A linear map version of `add_monoid_hom.eq_locus` -/
+def eq_locus (f g : M →ₛₗ[τ₁₂] M₂) : submodule R M :=
+{ carrier := {x | f x = g x},
+  smul_mem' := λ r x (hx : _ = _), show _ = _,
+    by simpa only [linear_map.map_smulₛₗ] using congr_arg ((•) (τ₁₂ r)) hx,
+  .. f.to_add_monoid_hom.eq_mlocus g.to_add_monoid_hom }
+
+@[simp] lemma mem_eq_locus {x : M} {f g : M →ₛₗ[τ₁₂] M₂} : x ∈ f.eq_locus g ↔ f x = g x :=
+iff.rfl
+
+lemma eq_locus_to_add_submonoid (f g : M →ₛₗ[τ₁₂] M₂) :
+  (f.eq_locus g).to_add_submonoid = f.to_add_monoid_hom.eq_mlocus g.to_add_monoid_hom :=
```

Comment:
Didn't we discuss on Zulip that the normal form should be to prefer the coercion over `to_add_monoid_hom`?

Ideally we'd also have a coercion for `submodule → add_submonoid`, but apparently that doesn't exist yet.
```suggestion
  (f.eq_locus g).to_add_submonoid = (f : M →+ M₂).eq_mlocus g :=
```